messenger.go

```diff
@@ -916,6 +935,111 @@ func (p *postProcessor) matchMessage(message *protocol.Message, chats []*Chat) (
 	}
 }
 
+type EnsDetails struct {
+	Name      string `json:"name"`
+	PublicKey string `json:"publicKey"`
+}
+
+type EnsResponse struct {
+	Name            string           `json:"name"`
+	Verified        bool             `json:"verified"`
+	Error           error            `json:"error"`
+	PublicKey       *ecdsa.PublicKey `json:"-"`
+	PublicKeyString string           `json:"publicKey"`
+}
+
+func (m *Messenger) verifyEnsName(ensInfo EnsDetails, ethclient *ethclient.Client, output chan EnsResponse) {
```

Comment:
If you want to keep this signature:
```suggestion
func (m *Messenger) verifyEnsName(ensInfo EnsDetails, ethclient *ethclient.Client, output chan<- EnsResponse) {
```